modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointManager.java

```diff
@@ -317,7 +317,18 @@ public void writePageToDeltaFilePageStore(
 
         CompletableFuture<DeltaFilePageStoreIo> deltaFilePageStoreFuture = filePageStore.getOrCreateNewDeltaFile(
                 index -> filePageStoreManager.tmpDeltaFilePageStorePath(pageId.groupId(), pageId.partitionId(), index),
-                () -> pageIndexesForDeltaFilePageStore(pagesToWrite.getPartitionView(pageMemory, pageId.groupId(), pageId.partitionId()))
+                () -> {
+                    CheckpointDirtyPagesView partitionView = pagesToWrite.getPartitionView(
```

Comment:
It got me thinking why it returns nullable at all, we could move assertion inside the method, but OKay